plutus/budget_manager/gcp_helper.py

```diff
@@ -0,0 +1,282 @@
+
+            metrics.incr("gcp_api_request_count", tags = ["type:billing.list", f"project_id:{project.project_id}"])
+        except Exception as err:
+            self.logger.error(f"Request to list budgets for {project.billing_account_id} failed due to: {err}.")
+            metrics.incr("error_count", tags = ["type:billing.list_budgets", f"project_id:{project.project_id}"])
+            sys.exit(1)
+
+        return budgets
+
+
+    def sync_config_with_gcp_budget(self, project, budget_dict, project_number):
+        '''
+        Returns two values.
+        1. Boolean whether configuration and gcp budget match.
+        2. A changed_budget object, with any diffs between config and GCP. Returns None if error.
+        '''
+
+        ret_val = True
+        # Make a deep copy, so tests can compare dicts more easily for changes
+        changed_budget = copy.deepcopy(budget_dict)
+
+        # Compare project id and project number to ensure we're dealing with the same budget.
+        if (f"projects/{project.project_id}" not in budget_dict['budget_filter']['projects'] and
+            f"projects/{project_number}" not in budget_dict['budget_filter']['projects']):
+            self.logger.error("Projectid and project number not found, this is an error and shouldnt happen, since we compare budget names prior")
+            metrics.incr("error_count", tags = ["type:misconfig", f"project_id:{project.project_id}"])
+            return False, None
+
+        # Ensure the budget is only for a singular project.
+        if len(budget_dict['budget_filter']['projects']) != 1:
+            self.logger.error("project ids list length, this shouldnt happen, we are going to assume one budget per project")
+            metrics.incr("error_count", tags = ["type:multiple_projectids"])
+            return False, None
+
+        # Compare and update budget
+        if project.budget_type == 'AMT':
+            if budget_dict['amount'].get('specified_amount') is None:
+                self.logger.info("Change detected from LASTMONTH to AMT")
+                ret_val = False
+                changed_budget['amount'] = { 'specified_amount': {'currency_code': 'USD', 'units': project.budget_amount } }
+
+            else:
+                # The GCP budget is already set to AMT, so we can rely on the other keys being there
+                if budget_dict['amount']['specified_amount']['currency_code'] != "USD":
+                    self.logger.error("non USD. This shouldnt happen.")
+                    metrics.incr("error_count", tags = ["type:non_usd", f"project_id:{project.project_id}"])
+                    return False, None
+
+                if project.budget_amount != budget_dict['amount']['specified_amount']['units']:
+                    self.logger.info("Change detected in budget units amount")
+                    ret_val = False
+                    changed_budget['amount']['specified_amount']['units'] = project.budget_amount
```

Comment:
I believe this could just be converted into two elif statements instead of these two if statements.